src/zjs_ipm.h

```diff
@@ -4,15 +4,27 @@
 #define __zjs_ipm_h__
 
 #include <ipm.h>
+#ifdef BUILD_MODULE_SENSOR
+#include <sensor.h>
+#endif
 
 #define IPM_CHANNEL_X86_TO_ARC                             0x01
 #define IPM_CHANNEL_ARC_TO_X86                             0x02
 
 // Message IDs
 #define MSG_ID_DONE                                        0x00
+#ifdef BUILD_MODULE_AIO
 #define MSG_ID_AIO                                         0x01
+#endif
+#ifdef BUILD_MODULE_I2C
 #define MSG_ID_I2C                                         0x02
+#endif
+#ifdef BUILD_MODULE_GROVE_LCD
 #define MSG_ID_GLCD                                        0x03
+#endif
+#ifdef BUILD_MODULE_SENSOR
+#define MSG_ID_SENSOR                                      0x04
+#endif
```

Comment:
I think you can go ahead and define all of these all the time, without the ifdefs. And actually I think elsewhere you use them without being in an ifdef so that would break the build anyway. Maybe review your usage of MSG_ID_* everywhere to make sure it makes sense.